relayer/workers/parachaincommitmentrelayer/ethereum-channel-writer.go

```diff
@@ -14,44 +14,45 @@ import (
 	"github.com/snowfork/polkadot-ethereum/relayer/chain"
 	"github.com/snowfork/polkadot-ethereum/relayer/chain/ethereum"
 	"github.com/snowfork/polkadot-ethereum/relayer/contracts/inbound"
-	"github.com/snowfork/polkadot-ethereum/relayer/substrate"
 )
 
 type EthereumChannelWriter struct {
-	config    *ethereum.Config
-	conn      *ethereum.Connection
-	contracts map[substrate.ChannelID]*inbound.Contract
-	messages  <-chan []chain.Message
-	log       *logrus.Entry
+	config                     *ethereum.Config
+	conn                       *ethereum.Connection
+	basicInboundChannel        *inbound.BasicInboundChannel
+	incentivizedInboundChannel *inbound.IncentivizedInboundChannel
```

Comment:
This makes the writer aware of specific channels – not sure that's a good design choice. Probably fine for now, having the writer agnostic to concrete channels might be a case of [YAGNI](https://xkcd.com/974/).

Reply:
This is very true. The previous version of the writer was agnostic to the concrete channels, because both channels conformed to the abstract `InboundChannel` contract and its `Message` struct. However this PR added a `fee` field to the `Message` struct of the incentivized inbound channel, and that meant the inbound channel could no longer conform to the `InboundChannel` interface.

This problem isn't isolated to the writer. The listener is also fully aware of the concrete channels (predating this PR).

So in future, I think what we need to explore is to add a generic `bytes data` to our Message structs. Then the incentivized channels can add have their fee info there, while still conforming to abstract `InboundChannel` and `OutboundChannel` contracts.

Reply:
@vgeddes makes sense. I'm also not sure this abstraction is necessary at this point, my inclination is rather to keep it simple until we need a multitude of channels in the future.

Reply:
ye sounds fine for now